code/CMSProfileController.php

```diff
@@ -2,113 +2,75 @@
 
 namespace SilverStripe\Admin;
 
-use SilverStripe\CMS\Controllers\CMSMain;
 use SilverStripe\Control\HTTPResponse;
 use SilverStripe\Forms\Form;
 use SilverStripe\Forms\HiddenField;
-use SilverStripe\Forms\FormAction;
-use SilverStripe\Model\List\ArrayList;
 use SilverStripe\Security\Member;
-use SilverStripe\Security\Permission;
 use SilverStripe\Security\Security;
 
-class CMSProfileController extends LeftAndMain
+class CMSProfileController extends SingleRecordAdmin
 {
     private static $url_segment = 'myprofile';
 
     private static $menu_title = 'My Profile';
 
     private static $required_permission_codes = 'CMS_ACCESS';
 
-    private static $tree_class = Member::class;
+    private static $model_class = Member::class;
 
     private static $ignore_menuitem = true;
 
-    public function getEditForm($id = null, $fields = null)
-    {
-        $this->setCurrentPageID(Security::getCurrentUser()->ID);
-
-        $form = parent::getEditForm($id, $fields);
-
-        if ($form instanceof HTTPResponse) {
-            return $form;
-        }
-
-        $form->Fields()->removeByName('LastVisited');
-        $form->Fields()->push(new HiddenField('ID', null, Security::getCurrentUser()->ID));
-        $form->Actions()->push(
-            FormAction::create('save', _t(CMSMain::class . '.SAVE', 'Save'))
-                ->addExtraClass('btn-primary font-icon-save')
-                ->setUseButtonTag(true)
-        );
-
-        $form->Actions()->removeByName('action_delete');
+    private static bool $only_one_record = false;
 
-        if ($member = Security::getCurrentUser()) {
-            $form->setValidator($member->getValidator());
-        } else {
-            $form->setValidator(Member::singleton()->getValidator());
-        }
+    private static bool $allow_new_record = false;
 
-        if ($form->Fields()->hasTabSet()) {
-            $form->Fields()->findOrMakeTab('Root')->setTemplate('SilverStripe\\Forms\\CMSTabSet');
+    protected function init()
+    {
+        parent::init();
+        if (!$this->getResponse()->isRedirect()) {
+            $this->setCurrentPageID(Security::getCurrentUser()->ID);
```

Comment:
Moving this into `init` instead of `getEditForm` so it takes affect for _all_ actions on this controller (including any that might be added through extensions).

Skipping for redirects because at that point we're not doing anything in this controller anyway.